src/queues.c

```diff
@@ -377,6 +377,40 @@ void queues_update(bool fullscreen)
 
                 iter = nextiter;
         }
+
+        /* If displayed is actually full, let the more important notifications
+         * from waiting seep into displayed.
+         */
+        if (settings.sort) {
+                GList *i_waiting, *i_displayed;
+
+                while (   (i_waiting   = g_queue_peek_head_link(waiting))
+                       && (i_displayed = g_queue_peek_tail_link(displayed))) {
+
+                        while (   fullscreen
```

Comment:
I'd break it down into a `queues_notification_ready` helper function to make this clearer.